compiler/sc1.cpp

```diff
@@ -3798,10 +3797,8 @@ static void dodelete()
  *                 | function-type-inner
  * function-type-inner ::= "function" type-expr "(" new-style-args ")"
  */
-static void parse_function_type(functag_t *type)
+static void parse_function_type(ke::UniquePtr<functag_t> &type)

```

Comment:
It looks like this signature didn't need to change? If so it should take a `functag_t*` or a `const ke::UniquePtr<functag_t>&`.